aqw-opcua-server/Settings.h

```diff
@@ -29,24 +29,20 @@ namespace weathersvr {
 
         private:
 
-            void setDefaultValues();
             /*
             Check if the values present in the settings.json file related to the dark sky api are valid before set them to respective variables. If is not valid, the default values will be kept.
             This function may throw an exception web::json::json_exception.
-            @param web::json::value& jsonObj - The Json OBJECT which the values will be parsed and validated.
+            @param web::json::value& jsonObj - The Json OBJECT which values will be parsed and validated.
             */
-            void validateValuesFromDarkSky(web::json::value& jsonObj);
+            bool validateValuesFromDarkSky(web::json::value& jsonObj);
 
             utility::string_t keyApiDarksky;
             utility::string_t units;
-            short intervalDownloadWeatherData;
+            int intervalWeatherDataDownload;
 
             //was public, changed to private
             int port_number;
             std::string endpointUrl;
             std::string hostName;
-
-            //Flag to track if we get invalid DarkSky API key from settings file. Can stop sending weather requests etc
-            bool validkeyApiDarksky = true;
```

Comment:
Can return `bool` flag but give it more generic name. There can be different reasons for `Settings` to be not valid (failed to open file, failed to parse Json, key is not valid one).